.github/workflows/automated-testing.yml

```
@@ -0,0 +1,351 @@
+        run: |
+          cp backend/.env.example backend/.env.test
+          sed -i 's/DATABASE_URL=.*/DATABASE_URL=postgresql:\/\/test:test@localhost:5432\/pachedu_test/' backend/.env.test
+          sed -i 's/NODE_ENV=.*/NODE_ENV=test/' backend/.env.test
+
+      - name: Run backend ${{ matrix.test-type }} tests
+        run: |
+          cd backend
+          npm test -- --testPathPattern="${{ matrix.test-type }}" --coverage --ci --watchAll=false
+
+      - name: Upload backend coverage
+        uses: codecov/codecov-action@v3
```

actionlint (1.7.7)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)